ci-operator/step-registry/ipi/conf/ibmcloud/ipi-conf-ibmcloud-commands.sh

```diff
@@ -40,10 +40,14 @@ esac
 # Select zone(s) based on REGION and ZONE_COUNT
 REGION="${LEASED_RESOURCE}"
 echo "ibmcloud region: ${REGION}"
-ZONES_COUNT=${ZONES_COUNT:-1}
-R_ZONES=("${REGION}-1" "${REGION}-2" "${REGION}-3")
-ZONES="${R_ZONES[*]:0:${ZONES_COUNT}}"
-ZONES_STR="[ ${ZONES// /, } ]"
+if [ -f "${SHARED_DIR}/dedicated_host.yaml" ]; then
+    ZONES_STR=$(yq-go r "${SHARED_DIR}/dedicated_host.yaml" "controlPlane.platform.ibmcloud.zones")
```

Comment:
`ipi-conf-ibmcloud-dedicatedhost` step can not overwrite the default ZONES_STR setting?

Reply:
with 'yq-go m -x -i "${CONFIG}" "${SHARED_DIR}/dedicated_host.yaml' fail to update the zones in install-config.yaml.

Reply:
You can delete the default zones setting firstly then merge it, something like:
```
[root@preserve-jialiu ~]# cat 1.install.yaml; echo "====="; yq-go d -i 1.install.yaml "controlPlane.platform.ibmcloud.zones"; cat 1.install.yaml; echo "==="; yq-go m -x -i 1.install.yaml 2.install.yaml; cat 1.install.yaml
basedomain: xx
controlPlane:
  platform:
    ibmcloud:
      type: ${masterProfile}
      zones: ["a", "b"]
      dedicatedHosts:
      - name: ${dhName_master}
compute:
- platform:
    ibmcloud:
      type: ${workerProfile}
      zones: ["c"]
      dedicatedHosts:
      - name: ${dhName_worker}
=====
basedomain: xx
controlPlane:
  platform:
    ibmcloud:
      type: ${masterProfile}
      dedicatedHosts:
      - name: ${dhName_master}
compute:
- platform:
    ibmcloud:
      type: ${workerProfile}
      zones: ["c"]
      dedicatedHosts:
      - name: ${dhName_worker}
===
basedomain: xx
controlPlane:
  platform:
    ibmcloud:
      type: ${masterProfile}
      dedicatedHosts:
      - name: ${dhName_master}
      zones:
      - "c"
compute:
- platform:
    ibmcloud:
      type: ${workerProfile}
      zones: ["b"]
      dedicatedHosts:
      - name: ${dhName_worker}
```
And you can do the operation in `ipi-conf-ibmcloud-dedicatedhost`, leave this step untouched.